tests/std/tests/P0896R4_ranges_alg_minmax/test.cpp

```diff
@@ -361,6 +363,68 @@ void test_gh_1893() {
     ASSERT(val == "meow");
 }
 
+class input_move_iterator {
+public:
+    using iterator_category = input_iterator_tag;
+    using iterator_concept  = input_iterator_tag;
+    using difference_type   = ptrdiff_t;
+    using value_type        = shared_ptr<int>;
+    using pointer           = shared_ptr<int>*;
+    using reference         = shared_ptr<int>&&;
+
+    input_move_iterator() = default;
+    input_move_iterator(shared_ptr<int>* ptr) : m_ptr(ptr) {}
+
+    reference operator*() const {
+        return ranges::iter_move(m_ptr);
+    }
+    pointer operator->() {
+        return m_ptr;
+    }
+
+    input_move_iterator& operator++() {
+        m_ptr++;
+        return *this;
+    }
+    input_move_iterator operator++(int) {
+        input_move_iterator tmp = *this;
+        ++(*this);
+        return tmp;
+    }
+
+    friend bool operator==(const input_move_iterator& a, const input_move_iterator& b) {
+        return a.m_ptr == b.m_ptr;
+    };
+    friend bool operator!=(const input_move_iterator& a, const input_move_iterator& b) {
+        return a.m_ptr != b.m_ptr;
+    };
+
+private:
+    shared_ptr<int>* m_ptr;
+};
+
+void test_gh_2900() {
+    // GH-2900: <algorithm>: ranges::minmax initializes minmax_result with the moved value
+    {
+        // check that the range access iterator isn't moved from multiple times
+        const string str{"this long string will be dynamically allocated"};
+        vector<string> v{str};
+        auto result = ranges::minmax(ranges::subrange{move_iterator{v.begin()}, move_iterator{v.end()}});
+        assert(result.min == str);
+        assert(result.max == str);
+    }
+    {
+        // check that the input iterator isn't moved from multiple times
+        shared_ptr<int> a[] = {make_shared<int>(42)};
+        auto range          = ranges::subrange(input_move_iterator(a), input_move_iterator(a + 1));

```

Comment:
Suggestion: It isn't exactly shadowing (because we're only `using namespace std;`), but this is named similarly to the concept `ranges::range`. Avoiding this by using `rng` would be nice. Additionally, we may as well use a named variable for the random access case above.